src/app/shared/components/pagination/pagination.component.ts

```diff
@@ -0,0 +1,30 @@
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { Observable, Subscription } from 'rxjs';
+
+@Component({
+  selector: 'hp-pagination',
+  templateUrl: './pagination.component.html',
+  styleUrls: ['./pagination.component.scss'],
+})
+export class PaginationComponent {
+  @Input() currentPage!: Observable<number>;
+  @Input() totalPages!: Observable<number>;
+  @Output() pageChanged: EventEmitter<number> = new EventEmitter();
+
+  subsription!: Subscription;
+  pageNumbers!: number[];
+
+  ngOnChanges(): void {
+    this.subsription = this.totalPages.subscribe(
+      (data) => (this.pageNumbers = Array.from({ length: data }, (_, i) => i))
+    );
+  }
```

Comment:
You override this.subsription with every change, so it seems like you make new subscribtion on every change, i suggest checking for existiong subscribtions. Also move pageNumbers calculation in separate method.

Reply:
I will rework ngFor in the template and remove this subscription